src/ios-deploy/ios-deploy.m

```diff
@@ -3113,6 +3113,28 @@ void kill_app(AMDeviceRef device) {
     [ns_pid release];
 }
 
+void launch_app(AMDeviceRef device) {
+    if (bundle_id == NULL ) {
+        on_error(@"Error: bundle id is required, please specify with --bundle_id");
+    }
+
+    instruments_connect_service(device);
+    instruments_perform_handshake();
+    
+    CFStringRef cf_bundle_id = CFAutorelease(CFStringCreateWithCString(NULL, bundle_id, kCFStringEncodingUTF8));
+    NSDictionary *options = @{@"StartSuspendedKey" : @NO, @"KillExisting" : @YES };
+        
+    int32_t channel = instruments_make_channel(@"com.apple.instruments.server.services.processcontrol");
+    
+    instruments_send_message(channel, @"launchSuspendedProcessWithDevicePath:bundleIdentifier:environment:arguments:options:", @[
```

Comment:
Should we use `instruments_perform_selector` instead and get the pid as the return value? Seems like this method can reply.
Reference: https://github.com/troybowman/dtxmsg/blob/master/dtxmsg_client.cpp#LL657C17-L657C17

Reply:
Yep, seems like we can do that. What might we want to do with this value? Print it to the console? Or just use it to check for errors?

Reply:
Printing makes sense to me. Useful in case someone wants to attach to the process later.